pkg/network/podinterfacenamescheme/podinterfacenamescheme.go

```diff
@@ -19,4 +19,37 @@
 
 package podinterfacenamescheme
 
+import (
+	"fmt"
+
+	v1 "kubevirt.io/api/core/v1"
+
+	"kubevirt.io/kubevirt/pkg/network/multus"
+)
+
 const PrimaryPodInterfaceName = "eth0"
+
+type PodInterfaceNamer map[string]string
+
+func NewPodInterfaceNamer(vmi *v1.VirtualMachineInstance) PodInterfaceNamer {
```

Comment:
How about `GenerateNameScheme`?

Reply:
I don't think it specific enough.

Reply:
The name "pod" embedded here assumes knowledge of who is using this and for what purpose.
I would argue it should not.

As I see it, this is a network ID generator. But it does not create a generator, it generates ID/s (or if you prefer, names).
The **caller** uses it to name the pod interface, when creating the CNI annotation or when consuming the pod interfaces.

Reply:
I would argue the main purpose is for pod interface names, but  don't mind using a more general name
how about `networkNameGenerator`?

Reply:
`networkNameGenerator` sounds like an object name, since this is a function I would suggest `generateNetworkName` instead.

Note that the package name suggests that it's all about pod interface names so IMO a more generic name can work too.